core/src/main/scala/akka/kafka/ConsumerSettings.scala

```diff
@@ -286,11 +288,12 @@ class ConsumerSettings[K, V](
     closeTimeout: FiniteDuration = closeTimeout,
     commitTimeout: FiniteDuration = commitTimeout,
     wakeupTimeout: FiniteDuration = wakeupTimeout,
+    maxWakeups: Int = maxWakeups,
     dispatcher: String = dispatcher
   ): ConsumerSettings[K, V] =
     new ConsumerSettings[K, V](properties, keyDeserializer, valueDeserializer,
       pollInterval, pollTimeout, stopTimeout, closeTimeout, commitTimeout, wakeupTimeout,
-      dispatcher)
+      maxWakeups, dispatcher)
```

Comment:
also add `withMaxWakups`

FYI: So far we don't care about breaking binary compatibility, but later we will and that is the reason why this is not a case class.